client/src/components/board-details/board-permissions/BoardPermissions.tsx

```diff
@@ -11,38 +11,53 @@ import { Table, TableBody, TableCell, TableRow } from "@/components/ui/table";
 import BoardPermissionsSelect from "@/components/board-details/board-permissions/BoardPermissionsSelect";
 import { BoardTableLeadRow } from "@/components/user-boards/board-table/BoardTableLeadRow";
 import DeleteCollaboratorButton from "@/components/board-details/board-permissions/DeleteCollaboratorButton";
-export const BoardPermissions = () => {
-  const [users, setUsers] = useState<BoardPermissionsUser[]>([
-    {
-      name: "Bnon Bnna",
-      permission: BoardPermission.MODERATOR,
-      addedAt: "2023-07-12 10:42 AM",
-      invitedBy: "Anon Anna",
-    },
-    {
-      name: "Cnon Cnna",
-      permission: BoardPermission.VIEWER,
-      addedAt: "yesterday 10:42 AM",
-      invitedBy: "Anon Anna",
-    },
-    {
-      name: "Dnon Dnna",
-      permission: BoardPermission.EDITOR,
-      addedAt: "yesterday 10:42 AM",
-      invitedBy: "Bnon Bnna",
-    },
-    {
-      name: "Enon Enna",
-      permission: BoardPermission.EDITOR,
-      addedAt: "yesterday 10:42 AM",
-      invitedBy: "Cnon Cnna",
-    },
-  ]);
+import { BoardDetailsResponse } from "@/interfaces/responses/board-details-response";
+import logger from "@/lib/logger";
+export const BoardPermissions = ({ board }: { board: BoardDetailsResponse }) => {
+  const mapPermissions = useCallback((board: BoardDetailsResponse): BoardPermissionsUser[] => {
+    const { viewer, editor, moderator } = board.permissions;
```

Comment:
`viewer`, `editor` and `moderator` are arrays, right? If so, I would change the name to something more like
```suggestion
    const { viewer: viewers, editor: editors, moderator: moderators } = board.permissions;
```

Reply:
done

Reply:
If we want to change that entirely, backend need to make some changes also